Lib/collections/__init__.py

```diff
@@ -454,12 +459,13 @@ def __getnewargs__(self):
     cache = _nt_itemgetters
     for index, name in enumerate(field_names):
         try:
-            itemgetter_object, doc = cache[index]
+            doc = cache[index]
         except KeyError:
-            itemgetter_object = _itemgetter(index)
             doc = f'Alias for field number {index}'
-            cache[index] = itemgetter_object, doc
-        class_namespace[name] = property(itemgetter_object, doc=doc)
+            cache[index] = doc
+
+        tuplegetter_object = _tuplegetter(index, doc=doc)
+        class_namespace[name] = tuplegetter_object
```

Comment:
Can tuplegetter object be cached?

```py
try:
    tuplegetter_object = cache[index]
except KeyError:
    tuplegetter_object = cache[index] = _tuplegetter(index, doc=f'Alias for field number {index}')
```

Reply:
Sadly, no because the docstrings are mutable. Check Serhiy comment:

https://github.com/python/cpython/pull/10495#discussion_r232916573